src/core/datastore/persisters/memory-key-value-persister.js

```diff
@@ -10,11 +10,12 @@ export class MemoryKeyValuePersister extends KeyValuePersister {
   }
 
   _readFromPersistance(key) {
-    return Promise.resolve(_storage[key]);
+    const result = _storage[key] || [];
+    return Promise.resolve(result.slice(0));
   }
 
   _writeToPersistance(key, array) {
-    _storage[key] = array;
+    _storage[key] = array.slice(0);
```

Comment:
Why was this change made?

Reply:
To avoid changing objects by reference. This was the reason one of the unit tests, which are currently skipped, was failing.

It was being called twice with the same array, except the second time it had another element in it. So at the time when the two calls to the spy were being compared, the array had 3 elements, instead of two. This is essentially how other persisters work too - when you read from localStorage, of instance, you always get a new object. Whereas here with inmemory persister, you got the same array each time you read. So it was being modified and everyone who has ever read it and kept it, got a modified array.